### DISCUSSION TITLE
docs: setsockopt(socket.SOL_SOCKET, 20, ... handler routine ...)

### DISCUSSION THREAD

**Discussion by massimosala** [micropython/micropython#11550] [ESP8266]
General question about TCP server, listen and accept client connections.

I don't recollect exactly, I am overloaded by too much infos in a few days.
Is the option 20 implemented for REPL over wss ?

If I don't use the remote REPL, can I use this option for my purposes?
Is it production ready ?
How many listen (servers) sockets can I use with this option?

---

If yes to the previous questions, can you please document its usage in the officlal docs?

**Comment by robert-hh** [comment]
This socket option was introduced for the ESP8266 port and it's webrepl feature. But it is not limited to that usage. YOu can use it for other purpose as well. It is for instance used in the ftp server (https://github.com/robert-hh/FTP-Server-for-ESP8266-ESP32-and-PYBD). Initially it was intended for use by ESP8266 only, but it's now available on most ports with network support.

**Comment by massimosala** [comment]
@robert-hh Hi Robert

Many thanks for your answer, and also for your help moving my other issues to the right place.

My goal: while developing and debugging I am used, also on linux and windows, to duplicate the usual print() also on another device, like a log file, a syslog server, etc... 

Due the few memory of some micros, I have this idea: make a minimal, unidirectional telnet server, where - if there is a client connected - it receives the debug print().

Sometimes it works, other times it disconnects the client a few second after the connections.
I have to take time and test it standalone, not with the main program... with about 2 Kb of free memory!

Can you please take a look at the attached code and see if there are major deficiences?
BTW, if it works and you feel it is a good idea, I will publish it.
Or it can be included in the official docs,  as an example :-)

Ciao, Massimo
[remote_print.py.txt](https://github.com/micropython/micropython/files/11511239/remote_print.py.txt)

**Reply by robert-hh** [reply]
You could compare your code with the telnet serber here, which uses the same mechanism. The part around swetting up the socket looks different. https://github.com/cpopp/MicroTelnetServer
